### PR TITLE
Remove deprecatedPublicIPs field

### PR DESCRIPTION
**Pull request by thockin**

### CODE REVIEW

**Comment by xiangpengzhao** [comment]
@thockin Did you generate `pkg/api/v1/types.generated.go` correctly? It's an empty file in your PR and it leads to the tests failure.

File: federation/apis/federation/v1beta1/types.generated.go [hunk 1]
```diff
@@ -1,2469 +0,0 @@
-/*
```
Comment (xiangpengzhao) [inline federation/apis/federation/v1beta1/types.generated.go]: Please check this generated file:)

File: examples/k8petstore/k8petstore.sh [hunk 2]
```diff
@@ -141,7 +141,7 @@ cat << EOF > fe-s.json
     "ports": [{
       "port": 3000
     }],
-    "deprecatedPublicIPs":["$PUBLIC_IP","10.1.4.89"],
+    "externalIPs":["$PUBLIC_IP","10.1.4.89"],
```
Comment (xiangpengzhao) [inline examples/k8petstore/k8petstore.sh]: Also change the name of `$PUBLIC_IP` to `EXTERNAL_IP` appears in this script?

**Comment by thockin** [comment]
@caesarxuchao I don't know what to do wrt staging updates.  I ran update-staging-client (update-all) and it made a bunch of changes like the following, which seem wrong.

```
@@ -15,11 +15,24 @@ go_library(
     ],
     tags = ["automanaged"],
     deps = [
+        "//pkg/apis/certificates:go_default_library",
+        "//pkg/apis/certificates/v1beta1:go_default_library",
         "//vendor/k8s.io/apimachinery/pkg/api/errors:go_default_library",
         "//vendor/k8s.io/apimachinery/pkg/apis/meta/v1:go_default_library",
         "//vendor/k8s.io/apimachinery/pkg/labels:go_default_library",
-        "//vendor/k8s.io/client-go/pkg/apis/certificates:go_default_library",
-        "//vendor/k8s.io/client-go/pkg/apis/certificates/v1beta1:go_default_library",
         "//vendor/k8s.io/client-go/tools/cache:go_default_library",
     ],
 )
```

**Comment by caesarxuchao** [comment]
The Jenkins verification passed anyway because we ignore the BUILD file diffs: https://github.com/kubernetes/kubernetes/blob/master/staging/copy.sh#L220-L221. @thockin if you run hack/update-bazel.sh again, are those changes cancelled?

I don't know which script is managing the BUILD files in the staging folder, @mikedanese do you know?

**Comment by thockin** [comment]
I manually reverted the obviously broken BUILD changes.

This is not a good developer experience.  I made a simple API change, I had
to run godep restore, manually revert BUILD files, and mess with my
GOPATH.  A casual contributor would have given up.  *I* almost gave up.

**Review by bowei — APPROVED** [review]

**Comment by mikedanese** [comment]
Gahh, that's a bug in how ./staging/copy.sh handles BUILD files, ./hack/update-bazel.sh is canonical. I'll take a look.

**Comment by caesarxuchao** [comment]
@mikedanese it seems the problem is exposed by https://github.com/kubernetes/kubernetes/pull/40777

**Comment by thockin** [comment]
Chao: the staging scripts assume they can call godep-restore, but without a
clean GOPATH that can fail in a hundred different ways, leaving the user
confused and frustrated.  It really needs to make a temporary space, change
GOPATH, restore into that, then process.  It sucks.  We should not be
asking users to do this.

**Comment by caesarxuchao** [comment]
@thockin the update-staging-client-go.sh [verifies](https://github.com/kubernetes/kubernetes/blob/master/hack/update-staging-client-go.sh#L24) the working dir is clean before proceeding. Or do you mean other things by "clean GOPATH"?

> It really needs to make a temporary space, change GOPATH, restore into that, then process

We had considered this, but it will take too long to run.

**Comment by thockin** [comment]
On Mon, Apr 17, 2017 at 12:44 PM, Chao Xu <notifications@github.com> wrote:
>
> @thockin the update-staging-client-go.sh verifies the working dir is clean before proceeding. Or do you mean other things by "clean GOPATH"?

If you have github.com/foobar in your own GOPATH, and then you do a
`godep restore` which includes that same package at a different rev it
can fail in a dozen different ways.

What I had to do to make it pass was:

mkdir _gopath
GOPATH=`pwd`/_gopath ./hack/godep-restore.sh
GOPATH=`pwd`/_gopath:$GOPATH ./hack/update-staging-client.sh

> It really needs to make a temporary space, change GOPATH, restore into that, then process
>
> We had considered this, but it will take too long to run.

The alternative is that it fails in hard to comprehend ways.  We
should not be making users do this at all.

**Comment by caesarxuchao** [comment]
> If you have github.com/foobar in your own GOPATH, and then you do a
`godep restore` which includes that same package at a different rev it
can fail in a dozen different ways.

`godep restore` fails in such an environment regardless of the staging folder. Currently `hack/update-staging-client.sh` and `hack/update-staging-godeps.sh` requires running `godep restore`. I target to eliminate the need for `update-staging-client.sh` this quarter. For `update-staging-godeps.sh`, I think @sttts once had a version of that script that operates in a temporary space, we can add it back as another alternative to running `hack/godep-restore.sh`, if it's easy.

In general, i think we should focus on eliminating the staging folder, instead of building more tools around it.

**Pull request merged**